subprojects/core/src/main/java/org/gradle/api/internal/tasks/userinput/UserInputHandler.java

```diff
@@ -16,12 +16,15 @@
 
 package org.gradle.api.internal.tasks.userinput;
 
+import org.gradle.api.provider.Provider;
 import org.gradle.internal.scan.UsedByScanPlugin;
+import org.gradle.internal.service.scopes.Scopes;
+import org.gradle.internal.service.scopes.ServiceScope;
 
 import javax.annotation.Nullable;
-import java.util.Collection;
 import java.util.function.Function;
 
+@ServiceScope(Scopes.Build.class)
```

Comment:
💭 It would be great to add a `reason` field to this annotation, so we could log the thinking behind each service existing in a certain scope.